test/distributed/_tensor/test_redistribute.py

```diff
@@ -175,6 +181,30 @@ def test_replicate_to_partial(self):
             partial_tensor.to_local(),
         )
 
+        # test backward path for replicate to partial where div should not
+        # be triggered.
+        local_tensor = torch.randn(12, 3, device=self.device_type)
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        partial_spec = DTensorSpec(
+            device_mesh,
+            (_Partial(),),
+        )
+        replica_spec = DTensorSpec(
+            device_mesh,
+            (Replicate(),),
+        )
+
+        out_local_forward = redistribute_local_tensor(
```

Comment:
Let's not test this API as it's sort of a private API, instead we just directly test `dtensor.from_local + redistribute`? I think it's already been tested in the above test too. If you want to just add some more assertions about the sizes of the gradients like you did here, we can remove the test here.

Reply:
Good catch! Removed this test in the latest version.